libs/hdf-converters/src/trufflehog-mapper.ts

```diff
@@ -0,0 +1,97 @@
+import {ExecJSON} from 'inspecjs';
+import _ from 'lodash';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {BaseConverter, ILookupPath, MappedTransform} from './base-converter';
+
+export class TrufflehogMapper extends BaseConverter {
+  withRaw: boolean;
+
+  mappings: MappedTransform<
+    ExecJSON.Execution & {passthrough: unknown},
+    ILookupPath
+  > = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion
+    },
+    version: HeimdallToolsVersion,
+    statistics: {},
+    profiles: [
+      {
+        name: {path: 'wrapper[0].SourceName'},
+        title: {path: 'wrapper[0].SourceName'},
+        supports: [],
+        attributes: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'wrapper',
+            key: 'id',
+            tags: {
+              nist: 'IA-5(7)',
+              cci: ['CCI-004069', 'CCI-000202', 'CCI-000203', 'CCI-002367'],
+              severity: 'medium'
+            },
+            refs: [],
+            source_location: {},
+            title: {
+              transformer: (data: Record<string, unknown>): string => {
+                return (
+                  _.get(data, 'DetectorName') + '_' + _.get(data, 'DecoderName')
+                );
+              }
+            },
+            id: {
+              transformer: (data: Record<string, unknown>): string => {
+                return (
+                  _.get(data, 'DetectorType') + '_' + _.get(data, 'DecoderName')
+                );
+              }
+            },
+            desc: {
+              transformer: (data: Record<string, unknown>): string => {
+                return (
+                  'Found ' +
+                  _.get(data, 'DetectorName') +
+                  ' secret using ' +
+                  _.get(data, 'DecoderName') +
+                  ' decoder'
+                );
+              }
+            },
+            impact: 0.5,
+            results: [
+              {
+                status: ExecJSON.ControlResultStatus.Failed,
+                code_desc: {
+                  transformer: (data: Record<string, unknown>): string => {
+                    return `${JSON.stringify(_.get(data, 'SourceMetadata'), null, 2)}`;
+                  }
+                },
+                message: {
```

Comment:
i really don't like this approach of just shoving everything into the message or code_desc.  here's the result struct (https://github.com/trufflesecurity/trufflehog/blob/55fe05d0b48a35bf3de294adde2b8aaee09a3004/pkg/output/json.go#L27) - it seems pretty consistent so we can make decisions on what to do with the data.  it allows for spaces that can include arbitrary data but we can treat those places appropriately.  

ex. if we're using the source name etc as the profile info, then we should probably not repeat all that jazz to infinity in here.  similarly if the detector information is being used to describe the control, then it doesn't need to all be repeated in the results area.  then there's the attributes that are empty strings or nulls: those can probably be removed from the output - just make sure to not exclude on all falsy values since there might be times like the 'verified' attribute where a true/false is an acceptable value.  i think only the following info should be retained in this message area (https://github.com/trufflesecurity/trufflehog/blob/55fe05d0b48a35bf3de294adde2b8aaee09a3004/pkg/output/json.go#L42-L53) and only if it meets the non-default/empty criteria as specified above.